pkg/admission/webhook/generic_webhook.go

```diff
@@ -79,15 +82,43 @@ func (c *clusterAwareSource) HasSynced() bool {
 var _ initializers.WantsKcpInformers = &WebhookDispatcher{}
 
 type WebhookDispatcher struct {
-	dispatcher              generic.Dispatcher
-	hookSource              ClusterAwareSource
-	apiBindingClusterLister apisv1alpha1listers.APIBindingClusterLister
-	apiBindingsHasSynced    cache.InformerSynced
 	*admission.Handler
+
+	dispatcher generic.Dispatcher
+	hookSource ClusterAwareSource
+
+	getAPIExport func(path logicalcluster.Path, name string) (*apisv1alpha1.APIExport, error)
+
+	apiBindingClusterLister apisv1alpha1listers.APIBindingClusterLister
+	apiExportIndexer        cache.Indexer
+
+	informersHaveSynced func() bool
+}
+
+func NewWebhookDispatcher() *WebhookDispatcher {
+	d := &WebhookDispatcher{
+		Handler: admission.NewHandler(admission.Connect, admission.Create, admission.Delete, admission.Update),
+	}
+
+	d.getAPIExport = func(path logicalcluster.Path, name string) (*apisv1alpha1.APIExport, error) {
+		objs, err := d.apiExportIndexer.ByIndex(indexers.ByLogicalClusterPathAndName, path.Join(name).String())
```

Comment:
Can't we use an APIExport lister?

Reply:
we can't. This is by path+name, not clusterName+name.